frontend/src/routes/Channels.tsx

```diff
@@ -0,0 +1,220 @@
+import { Form, Link, useLoaderData } from "react-router-dom";
+import { useEffect, useMemo, useState } from "react";
+import { Chat, User } from "../types/types";
+
+import { ColumnDef, createColumnHelper } from "@tanstack/react-table";
+
+import { Typography } from "../components/Typography";
+import Table from "../components/Table";
+import { Data } from "../data";
+import { Button } from "../components/Button";
+import { RootUser } from "./Root";
+import { Badge } from "../components/Badge";
+import { joinChannel } from "../actions/joinChannel";
+import { leaveChannel } from "../actions/leaveChannel";
+import { FiLock } from "react-icons/fi";
+import { Input } from "../components/Input";
+import { Card } from "../components/Card";
+import { Alert } from "../components/Alert";
+
+const columnHelper = createColumnHelper<Chat>();
+
+export default function Channels() {
+  const currentUser = RootUser();
+  const [allChats, userList] = useLoaderData() as [Chat[], User[]];
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [chatId, setChatId] = useState<number | undefined>(undefined);
+  const [password, setPassword] = useState("");
+  const [isError, setIsError] = useState(false);
+
+  const chats = allChats.filter((chat) => chat.type !== "direct");
+
+  const columns = useMemo<ColumnDef<Chat>[]>(
+    () => [
+      columnHelper.accessor("createdByUser", {
+        header: "Owner",
+        cell: (props) => {
+          let usr = "";
+          userList.forEach((user) => {
+            if (user.id == props.getValue()) {
+              usr = user.login;
+            }
+          });
+          return <span>{usr}</span>;
```

Comment:
Eu acho que se a relation for um User ao invés de um número não vai precisar isso - o props.getValue vai retornar o User aí dá pra pegar o login

Reply:
`userList `possue um array de Users, nao tem relacao com a relation da tabela, eu faco uma chamada direta para um dos endpoints de listar usuarios (que ja existia).